hs/src/Reach/JSUtil.hs

```diff
@@ -177,6 +178,12 @@ a2sp = JSSemi
 class HasJSAnnot a where
   jsa :: a -> JSAnnot
 
+class RepJSAnnot a where
+  rjsa :: JSAnnot -> a -> a
+
+rjsaJSL :: (HasJSAnnot a, RepJSAnnot a) => JSAnnot -> JSCommaList a -> JSCommaList a
+rjsaJSL a' y =  toJSCL $ map (rjsa a') $ jscl_flatten y
```

Comment:
I think with `RepJSAnnot`, you should just implement `class RepJSAnnot a => RepJSAnnot (JSCommaList a)`